### PR TITLE
`RaftPartition` can step down if it is not the primary

### LINKED ISSUE #8223: Allow raft to step down if it's not the expected leader
**Description**

Add a method `stepDownIfNotExpectedLeader` to RaftPartition that uses `PartitionMetadata` to decide if it should step down.
Expected leader = the member with the highest priority.

* If priority election is not enabled this method should have no effect.
* If fixed config is used, ensure that the priority is different for each nodes. If priorities are same for all nodes, then it is better to not stepdown as this will not guarantee expected leader distribution.

Related #6437

### PR DESCRIPTION
**Pull request by lenaschoenburg**
## Description

First step towards on-demand best-effort do-it-yourself leader rebalancing :tada: 
Here we add a method to `RaftPartition` that causes a stepdown if all conditions are met:
* Priority Election is enabled
* The partition distributor determined a primary node
* We are not the primary node 

## Related issues

closes #8223 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Review by deepthidevaki — CHANGES_REQUESTED** [review]
Great :rocket: 

Could you please also add  a test for Fixed distribution to verify primary is calculated correctly? How do you plan to test this feature? I was thinking whether we should add a unit test here or add an integration test once everything is implemented.

*

**Comment by oleschoenburg** [comment]
@deepthidevaki My apologies for rebasing while you are reviewing but there were some merge conflicts. I've added a test to make sure that the fixed distributor doesn't assign a leader if there is more than one potential leader. The other tests already implicitly verify that the correct leader is assigned.

As for testing the entire feature: I haven't found a good way to add a unit test but maybe I can pick your brain for some pointers.

File: broker/src/test/java/io/camunda/zeebe/broker/partitioning/distribution/FixedPartitionDistributorTest.java [hunk 1]
```diff
@@ -124,6 +128,33 @@ void shouldDistributeUnevenly() {
         .containsExactlyInAnyOrderElementsOf(expectedDistribution);
   }
 
+  @Test
+  void shouldNotAssignLeaderIfMoreThanOnePotentialLeader() {
```
Comment (deepthidevaki) [inline broker/src/test/java/io/camunda/zeebe/broker/partitioning/distribution/FixedPartitionDistributorTest.java:132]:
```suggestion
  void shouldNotAssignPrimaryIfMoreThanOnePotentialLeader() {
```

**Review by deepthidevaki — APPROVED** [review]
Thanks :+1:

I think an integration test with the full feature would be enough. So we can discuss it later.

File: broker/src/test/java/io/camunda/zeebe/broker/partitioning/distribution/FixedPartitionDistributorTest.java (hunk 1 above)
Comment (oleschoenburg) [reply broker/src/test/java/io/camunda/zeebe/broker/partitioning/distribution/FixedPartitionDistributorTest.java:132]: Oops, thanks :+1:

**Pull request merged**